pkg/test/framework/components/echo/kube/instance.go

```diff
@@ -205,3 +198,28 @@ func (c *instance) Restart() error {
 		return nil
 	}, retry.Timeout(c.cfg.ReadinessTimeout), startDelay)
 }
+
+// aggregateResponses forwards an echo request from all workloads belonging to this echo instance and aggregates the results.
+func (c *instance) aggregateResponses(opts echo.CallOptions, retry bool, retryOptions ...retry.Option) (appEcho.ParsedResponses, error) {
```

Comment:
Should we only aggregate across different versions? In other words, if running with a single version, do exactly the same thing we do today?

Reply:
Makes sense. I guess the case this behaves differently with multiple `Subset`s, or are there other places where we expect multiple workloads per instance?

Reply:
That's a good question. Maybe we should have all workloads in the same instance attached to the same control plane ... probably simpler that way. That would mean that we'd want at least 6 echo instances per test (2 per control plane version, so we test all combinations of N, N-1, N-2). Make sense?